samples/secure-fraud-detection/README.md

```diff
@@ -0,0 +1,118 @@
+
+```java
+@RegisterAiService(retrievalAugmentor = FraudDetectionRetrievalAugmentor.class)
+```
+
+For each message, the prompt is engineered to help the LLM understand the context and answer the request:
+
+```java
+    @SystemMessage("""
+            You are a bank account fraud detection AI. You have to detect frauds in transactions.
+            """)
+    @UserMessage("""
+            Your task is to detect whether a fraud was committed for the customer.
+
+            Answer with a **single** JSON document containing:
+            - the customer name in the 'customer-name' key
+            - the transaction limit in the 'transaction-limit' key
+            - the computed sum of all transactions committed during the last 15 minutes in the 'total' key
+            - the 'fraud' key set to true if the computed sum of all transactions is greater than the transaction limit
+            - the 'transactions' key containing an array of JSON objects. Each object must have transaction 'amount', 'city' and formatted 'time' keys.
+            - the 'explanation' key containing an explanation of your answer.
+            - the 'email' key containing the customer email if the fraud was detected. 
+            
+           Your response must be just the raw JSON document, without ```json, ``` or anything else. Do not use null JSON properties.
+            """)
+@Timeout(value = 2, unit = ChronoUnit.MINUTES)
+String detectAmountFraudForCustomer();
+```
+
+_Note:_ You can also use fault tolerance annotations in combination with the prompt annotations.
+
+### Using the AI service
+
+Once defined, you can inject the AI service as a regular bean, and use it:
+
+```java
+package io.quarkiverse.langchain4j.sample;
+
+import io.quarkus.security.Authenticated;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+@Path("/fraud")
+@Authenticated
+public class FraudDetectionResource {
+
+    private final FraudDetectionAi service;
+
+    public FraudDetectionResource(FraudDetectionAi service) {
+        this.service = service;
+    }
+
+    @GET
+    @Path("/amount")
+    public String detectBaseOnAmount() {
+    	return service.detectAmountFraudForCustomer();
+    }
+}
+```
+
+`FraudDetectionResource` can only be accessed by authenticated users.
+
+## Google Authentication
+
+This demo requires users to authenticate with Google.
+All you need to do is to register an application with Google, follow steps listed in the [Quarkus Google](https://quarkus.io/guides/security-openid-connect-providers#google) section.
+Name your Google application as `Quarkus LangChain4j AI`, and make sure an allowed callback URL is set to `http://localhost:8080/login`.
+Google will generate a client id and secret, use them to set `quarkus.oidc.client-id` and `quarkus.oidc.credentials.secret` properties.
+
+## Running the Demo
+
+To run the demo, use the following command:
+
+```shell
+mvn quarkus:dev -Dname="Firstname Familyname" -Demail=someuser@gmail.com
```

Comment:
Can we be even a bit more explicit that the `name` and `email` have to exactly match the user's google account (maybe say that this is used to automatically add a user entry into the app's database)?

Reply:
@jmartisk This is all mentioned in the Setup section above, starting from line 10:

```
The demo requires that your Google account's full name and email are configured.
You can use system or env properties, see `Running the Demo` section below.
```

following the info that transactions are generated.

I think users will understand the exact match is required. Can you suggest some further edits please there ?

Note, if they made a mistake during the registration, the error page will show them what exactly they should register by listing a full name and email, I added that page yesterday.